component/configsource.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component // import "go.opentelemetry.io/collector/component"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configmapprovider"
+)
+
+// ConfigSource is a component that provides a configuration map or value.
+type ConfigSource interface {
```

Comment:
Did you see https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/internal/configsource and https://github.com/open-telemetry/opentelemetry-collector/tree/main/component/experimental/component

Reply:
Yes, I saw it. It will be all deleted since it is no longer needed. I deleted some parts of it already in this PR (the Manager).

Reply:
Is this feature can be achieved through helm chart